lib/journaled/outbox/adapter.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+module Journaled
+  module Outbox
+    # Outbox-style delivery adapter for custom event processing
+    #
+    # This adapter stores events in a database table instead of enqueuing to ActiveJob.
+    # Events are processed by separate worker daemons that poll the database.
+    #
+    # Setup:
+    # 1. Generate migrations: rails generate journaled:database_events
+    # 2. Run migrations: rails db:migrate
+    # 3. Configure: Journaled.delivery_adapter = Journaled::Outbox::Adapter
+    # 4. Start workers: bundle exec rake journaled_worker:work
+    class Adapter < Journaled::DeliveryAdapter
+      class TableNotFoundError < StandardError; end
+
+      # Delivers events by inserting them into the database
+      #
+      # @param events [Array] Array of journaled events to deliver
+      # @param ** [Hash] Additional options (ignored, for interface compatibility)
+      # @return [void]
+      def self.deliver(events:, **)
+        check_table_exists!
+
+        events.each do |event|
+          # Exclude the application-level id - the database will generate its own using uuid_generate_v7()
+          event_data = event.journaled_attributes.except(:id)
+
+          Event.create!(
+            event_type: event.journaled_attributes[:event_type],
+            event_data:,
+            partition_key: event.journaled_partition_key,
+            stream_name: event.journaled_stream_name,
+          )
+        end
+      end
+
+      # Check if the required database table exists
+      #
+      # @raise [TableNotFoundError] if the table doesn't exist
+      def self.check_table_exists!
+        return if table_exists_cache
+
+        unless Event.table_exists?
+          raise TableNotFoundError, <<~ERROR
+            Journaled::Outbox::Adapter requires the 'journaled_outbox_events' table.
+
+            To create the required tables, run:
+
+              rails generate journaled:database_events
+              rails db:migrate
+
+            For more information, see the README:
+            https://github.com/Betterment/journaled#outbox-style-event-processing-optional
+          ERROR
+        end
+
+        self.table_exists_cache = true
+      end
```

Comment:
```suggestion
      def self.check_table_exists!
        return if @table_exists

        unless Event.table_exists?
          raise TableNotFoundError, <<~ERROR
            Journaled::Outbox::Adapter requires the 'journaled_outbox_events' table.

            To create the required tables, run:

              rails generate journaled:database_events
              rails db:migrate

            For more information, see the README:
            https://github.com/Betterment/journaled#outbox-style-event-processing-optional
          ERROR
        end

        @table_exists = true
      end
```

I know we like to avoid instance variables, but if they never need to be referenced outside of a single method, I kinda don't mind, and it avoids having a code locality problem with the attr_accessor.